x-pack/plugins/maps/public/routing/routes/list/list.js

```diff
@@ -0,0 +1,451 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import _ from 'lodash';
+import { getMapsSavedObjectLoader } from '../../../bootstrap/services/gis_map_saved_object_loader';
+import {
+  getMapsCapabilities,
+  getUiSettings,
+  getToasts,
+  getCoreChrome,
+} from '../../../kibana_services';
+import {
+  EuiTitle,
+  EuiFieldSearch,
+  EuiBasicTable,
+  EuiPage,
+  EuiPageBody,
+  EuiPageContent,
+  EuiLink,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiSpacer,
+  EuiOverlayMask,
+  EuiConfirmModal,
+  EuiCallOut,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { addHelpMenuToAppChrome } from '../../../help_menu_util';
+import { Link, withRouter } from 'react-router-dom';
+import { updateBreadcrumbs } from '../../page_elements/breadcrumbs';
+import { goToSpecifiedPath } from '../../maps_router';
+
+export const EMPTY_FILTER = '';
+
+export const MapsListView = withRouter(
+  class extends React.Component {
+    state = {
+      hasInitialFetchReturned: false,
+      isFetchingItems: false,
+      showDeleteModal: false,
+      showLimitError: false,
+      filter: EMPTY_FILTER,
+      items: [],
+      selectedIds: [],
+      page: 0,
+      perPage: 20,
+      readOnly: !getMapsCapabilities().save,
+      listingLimit: getUiSettings().get('savedObjects:listingLimit'),
+    };
+
+    UNSAFE_componentWillMount() {
+      this._isMounted = true;
+      updateBreadcrumbs();
+    }
+
+    componentWillUnmount() {
+      this._isMounted = false;
+      this.debouncedFetch.cancel();
+    }
+
+    async componentDidMount() {
+      this.fetchItems();
+      addHelpMenuToAppChrome();
+      getCoreChrome().docTitle.change('Maps');
+    }
+
+    _find = (search) => getMapsSavedObjectLoader().find(search, this.state.listingLimit);
+
+    _delete = (ids) => getMapsSavedObjectLoader().delete(ids);
+
+    debouncedFetch = _.debounce(async (filter) => {
```

Comment:
Leaving as is for now since it's not a change from this PR. I think this whole file could be broken up and changed quite a bit but it's an issue for future work